src/plugins/intel_gpu/src/plugin/ops/pooling.cpp

```diff
@@ -45,6 +42,38 @@ static void CreateAvgPoolOp(ProgramBuilder& p, const std::shared_ptr<ov::op::v1:
     p.add_primitive(*op, pooling_prim);
 }
 
+static void CreateAvgPoolOp(ProgramBuilder& p, const std::shared_ptr<ov::op::v16::AvgPool>& op) {
```

Comment:
I think we can try to make a general function for creating a primitive
`AvgPoolImpl(ProgramBuilder& p, const std::shared_ptr<ov::Node>& op..`
And call it with different parameters from overloaded factories
For example: src/plugins/intel_gpu/src/plugin/ops/lora.cpp